src/Data/JSON/Fields.hs

```diff
@@ -46,6 +48,15 @@ instance Apply ToJSONFields1 fs => ToJSONFields1 (Sum fs) where
 instance (ToJSONFields a, ToJSONFields b) => ToJSONFields (a, b) where
   toJSONFields (a, b) = [ "before" .= JSONFields a, "after" .= JSONFields b ]
 
+instance ToJSONFields Range where
+  toJSONFields Range{..} = ["sourceRange" .= [ start, end ]]
+
+instance ToJSONFields Span where
+  toJSONFields sourceSpan = [ "sourceSpan" .= sourceSpan ]
+
+instance ToJSONFields Loc where
+  toJSONFields Loc{..} = toJSONFields byteRange <> toJSONFields span
```

Comment:
All these instances live here now so we didn’t have to move `ToJSONFields` into `semantic-source` or anything.